packages/core-bridge/src/client.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{
+    conversions::*,
+    errors::*,
+    runtime::{BoxedRuntimeRef, FutureToPromise, RuntimeHandle},
+};
+use neon::{context::Context, prelude::*};
+use std::{cell::RefCell, sync::Arc};
+use temporal_client::{ClientInitError, ConfiguredClient, TemporalServiceClientWithMetrics};
+use temporal_sdk_core::RetryClient;
+
+use neon::{
+    handle::Handle,
+    types::{JsBoolean, JsNumber, JsString},
+};
+use std::{collections::HashMap, time::Duration};
+use temporal_client::HttpConnectProxyOptions;
+use temporal_sdk_core::{
+    ClientOptions, ClientOptionsBuilder, ClientTlsConfig, RetryConfig, TlsConfig, Url,
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// This is the type that we actually pass to the lang side.
+///
+/// - JsBox: So that we're informed if the object is dropped by the lang GC
+/// - RefCell: For interior mutability
+/// - Option: So that we can take it out of the box on close (requires mutability ^^^)
+/// - Arc: So that we can safely pass the ClientHandle around -- FIXME: Is this useful?
+/// - ClientHandle: The actual bridge client handle (below)
+pub type BoxedClientRef = JsBox<RefCell<Option<ClientHandle>>>;
+
+pub struct ClientHandle {
+    pub(crate) runtime_handle: Arc<RuntimeHandle>, // FIXME: Should we inline rather than Arc?
```

Comment:
There's no reason to Arc here since runtime handle itself is using them internally.